xarray/core/duck_array_ops.py

```diff
@@ -171,6 +171,79 @@ def _ignore_warnings_if(condition):
         yield
 
 
+def _nansum(value, axis=None, **kwargs):
+    """ In house nansum for object array """
+    value = fillna(value, 0.0)
+    return _dask_or_eager_func('sum')(value, axis=axis, **kwargs)
+
+
+def _nan_minmax(func, fill_value, value, axis=None, **kwargs):
+    """ In house nanmin and nanmax for object array """
+    valid_count = count(value, axis=axis)
+    filled_value = fillna(value, fill_value)
+    data = _dask_or_eager_func(func)(filled_value, axis=axis, **kwargs)
+    if not hasattr(data, 'dtype'):  # scalar case
+        data = np.nan if valid_count == 0 else data
```

Comment:
Instead passing in the `fill_value`, let's figure it out from `dtypes.maybe_promote()`